interface/src/app/mlmodel/resource.js

```diff
@@ -0,0 +1,9 @@
+angular.module('adage.mlmodel.resource', ['ngResource'])
+
+.factory('MLModel', ['$resource', function($resource) {
+  return $resource(
+    '/api/v0/mlmodel/'
+  );
+}])
```

Comment:
Aha. I hadn't seen @dongbohu's comment when I wrote mine.

Architecturally, it is important to maintain the loose coupling @rzelayafavila describes. This is another example of separation of concerns: the UI code should not have direct knowledge of API endpoints but rather work their magic through a collection of thoughtfully managed services.

This is the same logic as abstracting database calls away from the application code as we do with Django's abstract model design. While there may be a small performance penalty involved, what we are optimizing for is maintainability and *programmer* efficiency. Slow running code is a concern too, of course, but that comes in at a distant second priority to readable and maintainable code.